src/shogun/base/SGObject.cpp

```diff
@@ -803,7 +803,9 @@ CSGObject* CSGObject::get(const std::string& name, index_t index) const
 CSGObject* CSGObject::get(std::string_view name, index_t index) const
 #endif
 {
-	auto* result = sgo_details::get_by_tag(this, name, sgo_details::GetByNameIndex(index));
+	//auto* result = sgo_details::get_by_tag(this, name, sgo_details::GetByNameIndex(index));
+	auto* result = sgo_details::get_by_tag(this, name);
```

Comment:
the problem here that imo this ignores the `index` parameter. so the visitor that you've written only works in the expected manner for `CSGObject::get(std::string_view name, std::nothrow_t)`. i need to see how we support get `CSGObject* CSGObject::get(std::string_view name, index_t index) const`. that's basically when the registered parameter is an array